service/src/test/java/com.codeforcommunity/processor/ReservationProcessorImplTest.java

```diff
@@ -0,0 +1,330 @@
+      fail("Method should not have thrown an exception");
+    }
+
+    try {
+      this.proc.isAdminCheck(PrivilegeLevel.SUPER_ADMIN);
+    } catch (AuthException e) {
+      fail("Method should not have thrown an exception");
+    }
+  }
+
+  @Test
+  public void testBlockOpenCheckNoAction() {
+    setup();
+    doReturn(Optional.empty()).when(this.proc).lastAction(anyInt());
+
+    try {
+      this.proc.blockOpenCheck(1);
+    } catch (IncorrectBlockStatusException e) {
+      fail("Method should not throw exception");
+    }
+  }
+
+  @Test
+  public void testBlockOpenCheckUncomplete() {
+    setup();
+    ReservationsRecord mockReservation = mock(ReservationsRecord.class);
+    doReturn(Optional.of(mockReservation)).when(this.proc).lastAction(anyInt());
+
+    when(mockReservation.getActionType())
+        .thenReturn(ReservationAction.UNCOMPLETE, ReservationAction.UNCOMPLETE);
+
+    try {
+      this.proc.blockOpenCheck(1);
+    } catch (IncorrectBlockStatusException e) {
+      fail("Method should not throw exception");
+    }
+  }
+
+  @Test
+  public void testBlockOpenCheckReserved() {
+    setup();
+    ReservationsRecord mockReservation = mock(ReservationsRecord.class);
+    doReturn(Optional.of(mockReservation)).when(this.proc).lastAction(anyInt());
+
+    when(mockReservation.getActionType())
+        .thenReturn(ReservationAction.RESERVE, ReservationAction.RESERVE);
+
+    try {
+      this.proc.blockOpenCheck(1);
+      fail();
+    } catch (IncorrectBlockStatusException e) {
+      assertEquals(1, e.getBlockId());
+      assertEquals("open", e.getExpectedStatus());
+    }
+  }
+
+  @Test
+  public void testBlockReservedCheckNoLastActionFailure() {
+    setup();
+    doReturn(Optional.empty()).when(this.proc).lastAction(anyInt());
+
+    try {
+      this.proc.blockReservedCheck(1, 15);
+      fail();
+    } catch (IncorrectBlockStatusException e) {
+      assertEquals(1, e.getBlockId());
+      assertEquals("reserved", e.getExpectedStatus());
+    }
+  }
+
+  @Test
+  public void testBlockReservedCheckWrongActionTypeFailure() {
+    setup();
+    ReservationsRecord mockReservation = mock(ReservationsRecord.class);
+    doReturn(Optional.of(mockReservation)).when(this.proc).lastAction(anyInt());
+    doReturn(false).when(this.proc).isOnTeam(anyInt(), anyInt());
+
+    when(mockReservation.getActionType()).thenReturn(ReservationAction.COMPLETE);
+    when(mockReservation.getUserId()).thenReturn(2, 2);
+    when(mockReservation.getTeamId()).thenReturn(3, 3);
+
+    try {
+      this.proc.blockReservedCheck(1, 15);
+      fail();
+    } catch (IncorrectBlockStatusException e) {
+      assertEquals(1, e.getBlockId());
+      assertEquals("reserved", e.getExpectedStatus());
+    }
+
+    verify(mockReservation, times(1)).getActionType();
+    verify(mockReservation, times(0)).getUserId();
+    verify(mockReservation, times(0)).getTeamId();
+    verify(this.proc, times(0)).isOnTeam(anyInt(), anyInt());
```

Comment:
Instead of doing `times(0)`, it might be better to do `never()`

Reply:
Probably but I liked the consistent format

Reply:
Sure